src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageLoaderResult.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure
+{
+    public struct PageLoaderResult
```

Comment:
The issue here is the `/_ViewImports.cshtml` (the one above the `/Pages` directory. For (c) we say `GetHierarchicalItems('${PagesRoot}/Index.cshtml")` and watch all those files along with $"{PagesRoot}/**/*.cshtml"